TP-Sampler/src/ui.js

```diff
@@ -383,6 +381,16 @@ function appendUIHandlers(addVariable, removeVariable, addVariableSeries, runBut
 
   let keyPressed = false;
 
+  document.getElementById("device_name").addEventListener("blur", function (e) {
+    setDeviceName(e.target.value);
+  });
+
+  document.getElementById("device_name").addEventListener("keydown", function (e) {
+    if (e.keyCode === 13) {
```

Comment:
Just Enter and not tab?

Reply:
yes, for the device name input it should only be when the user hits 'enter' or clicks off of the input